docker/docker.ts

```diff
@@ -315,28 +343,25 @@ async function dockerBuild(
         buildArgs.push(...[ "--target", target ]);
     }
 
-    const buildResult = await runCLICommand("docker", buildArgs, logResource, /*reportStdout*/ true);
-    if (buildResult.code) {
-        throw new ResourceError(`Docker build of image '${imageName}' failed with exit code: ${buildResult.code}`, logResource);
-    }
+    await runCommandThatMustSucceed("docker", buildArgs, logResource);
```

Comment:
instead of having to check the code and throw, this is taken care of for you automatically by runCommandThatMustSucceed.  if runCommandThatMustSucceed hits a failure, it automatically makes a good error message (including full command output) and throws it so that we'll properly report things later.